ha/resource/dynamic_fid_service_ra.py

```diff
@@ -0,0 +1,267 @@
+            service (str): Service name
+
+        Returns:
+            str: Return service status.
+                Service status are one of failed, active, unknown, activating
+        """
+        output, _, _ = self._execute.run_cmd(f"systemctl is-active {service}",
+                                                check_error=False)
+        return output
+
+    def metadata(self) -> int:
+        """
+        Provide meta data for resource agent and parameter
+        """
+        env: str =r"""<?xml version="1.0"?>
+        <!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+        <resource-agent name="dynamic_fid_service_ra">
+        <version>1.0</version>
+
+        <longdesc lang="en">
+        This is resource agent, wrapper around systemd service.
+        It map clone to service instance on node.
+        </longdesc>
+        <shortdesc lang="en">Systemd wrapper agent</shortdesc>
+        <parameters>
+        <parameter name="service" required="0">
+        <longdesc lang="en"> Service name to manage systemd </longdesc>
+        <shortdesc lang="en"> Systemd service </shortdesc>
+        <content type="string"/>
+        </parameter>
+        <parameter name="fid_service_name" required="0">
+        <longdesc lang="en"> Fid service name used in mapping </longdesc>
+        <shortdesc lang="en"> Systemd service </shortdesc>
+        <content type="string"/>
+        </parameter>
+        </parameters>
+        <actions>
+        <action name="start"        timeout="40s" />
+        <action name="stop"         timeout="40s" />
+        <action name="monitor"      timeout="40s" interval="60s" depth="0" />
+        <action name="meta-data"    timeout="4s" />
+        </actions>
+        </resource-agent>
+        """
+        sys.stdout.write(env)
+        return const.OCF_SUCCESS
+
+    def start(self) -> int:
+        """
+        Start service and provide output.
+
+        Command to start service:
+            $ systemctl reset-failed service
+            $ systemctl start service
+
+        Returns:
+            int: Return as per service status.
+                active: return const.OCF_SUCCESS.
+                unknown: Wait till timeout.
+                failed or timeout will cause failover or moved to Stopped state.
+        """
+        service = self._get_systemd_service()
+        Log.debug(f"Start: Start {service} service")
+        self._execute.run_cmd(f"systemctl reset-failed {service}", check_error=False)
+        self._execute.run_cmd(f"systemctl start {service}", check_error=False)
+        while True:
+            Log.debug(f"Start: Starting {service} service")
+            status: str = self._get_service_status(service).strip()
+            if status == "active":
+                break
+            elif status == "failed":
+                Log.info(f"Start: Failed to start {service} and may cause failover or Stop.")
+                return const.OCF_ERR_GENERIC
+            else:
+                time.sleep(1)
```

Comment:
I am not sure if the sleep here and in below functions is the right thing to do. I believe we should check the status and return immediately.

Reply:
we will need setup
For start
systemctl start is async command will not check if service is started
even if we use systemctl status 
some service take time to start so status will not be active till it start
so during this time we need to check again 
and only return if service started.

Reply:
ok